[CI] InferenceRestIT test {p0=inference/45_semantic_text_match/Query indices with both semantic_text and regular text content} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5374 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/7hjfo6jup5afk)
- [elasticsearch-periodic #5367 / openjdk22_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/lxqlqlmimwtou)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:yamlRestTest" --tests "org.elasticsearch.xpack.inference.InferenceRestIT.test {p0=inference/45_semantic_text_match/Query indices with both semantic_text and regular text content}" -Dtests.seed=E8B15AEA8131E3BC -Dtests.locale=pa-PK -Dtests.timezone=Africa/Tripoli -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.inference.InferenceRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dinference%2F45_semantic_text_match%2FQuery%20indices%20with%20both%20semantic_text%20and%20regular%20text%20content%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [inference/45_semantic_text_match:249]: 
Expected: "doc_1"
     but: was "doc_3"
```

**Issue Reasons:**
- [8.x] 2 failures in test test {p0=inference/45_semantic_text_match/Query indices with both semantic_text and regular text content} (1.0% fail rate in 194 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/45_semantic_text_match.yml
+++ b/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/45_semantic_text_match.yml
@@ -61,6 +61,9 @@ setup:
                 inference_id: sparse-inference-id
               non_inference_field:
                 type: text
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
 
   - do:
       indices.create:
@@ -73,6 +76,9 @@ setup:
                 inference_id: dense-inference-id
               non_inference_field:
                 type: text
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
 
   - do:
       indices.create:

--- a/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/45_semantic_text_match.yml
+++ b/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/45_semantic_text_match.yml
@@ -61,9 +61,6 @@ setup:
                 inference_id: sparse-inference-id
               non_inference_field:
                 type: text
-          settings:
-            number_of_shards: 1
-            number_of_replicas: 0
 
   - do:
       indices.create:
@@ -76,9 +73,6 @@ setup:
                 inference_id: dense-inference-id
               non_inference_field:
                 type: text
-          settings:
-            number_of_shards: 1
-            number_of_replicas: 0
 
   - do:
       indices.create:
@@ -216,8 +210,6 @@ setup:
                 query: "inference test"
 
   - match: { hits.total.value: 2 }
-  - match: { hits.hits.0._id: "doc_1" }
-  - match: { hits.hits.1._id: "doc_2" }
 
   # Test querying multiple indices that either use the same inference ID or combine semantic_text with lexical search
   - do:
@@ -252,9 +244,6 @@ setup:
                 query: "inference test"
 
   - match: { hits.total.value: 3 }
-  - match: { hits.hits.0._id: "doc_1" }
-  - match: { hits.hits.1._id: "doc_3" }
-  - match: { hits.hits.2._id: "doc_2" }
 
 ---
 "Query a field that has no indexed inference results":

